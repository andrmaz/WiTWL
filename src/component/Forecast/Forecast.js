import React from 'react'
import PropTypes from 'prop-types'
import './Forecast.scss';

function Forecast({ weather }) {
    return (
        <section className='box4'>
            <h1 className='title'>Forecast</h1> 
            <div className='forecast'>
                {
                    weather?.forecast?.forecastday.map((day, index) => (
                        <div key={index} className='card'>
                            <p>{day.day.condition.text}</p>
                            <span>{`Date: ${day.date}`}</span>
                            <img 
                                src={`http:${day.day.condition.icon}`} 
                                alt='forecastday'
                            />
                        </div>
                    ))
                }
            </div>
        </section>
    )
}

Forecast.propTypes = {
    weather: PropTypes.object
}

export default Forecast

