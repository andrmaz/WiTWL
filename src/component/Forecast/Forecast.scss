.box4 {
    
    grid-column-start: 1;
    grid-column-end: 3;

    .title {
        height: 10%;
        text-align: center;
    }
    .forecast {
        height: 90%;
        display: flex;
        justify-content: space-around;
        align-items: center;

        .card {
            height: 80%;
            width: 30%;
            display: flex;
            flex-direction: column;
            justify-content: space-evenly;
            align-items: center;
        }
    }
}