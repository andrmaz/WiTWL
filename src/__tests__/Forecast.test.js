import React from 'react';
import { shallow } from 'enzyme';
import Forecast from '../component/Forecast/Forecast';

let wrapper
const mockWeather = {
    forecast: { 
        forecastday: [{
            day: {
                condition: '',
                text: ''
            },
            date: ''
        }]
    }
}

describe('Forecast Component', () => {
    beforeEach(() => {
        wrapper = shallow(<Forecast weather={mockWeather}/>)
    })
    it('should render component correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    it('should render forecast correctly', () => {
        expect(mockWeather.forecast.forecastday).toHaveLength(1)
    })
})