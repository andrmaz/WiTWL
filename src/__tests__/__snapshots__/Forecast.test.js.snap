// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Component should render component correctly 1`] = `
<section
  className="box4"
>
  <h1
    className="title"
  >
    Forecast
  </h1>
  <div
    className="forecast"
  >
    <div
      className="card"
      key="0"
    >
      <p />
      <span>
        Date: 
      </span>
      <img
        alt="forecastday"
        src="http:undefined"
      />
    </div>
  </div>
</section>
`;
